src/chain/auth.sw

```diff
@@ -1,28 +1,43 @@
 library auth;
+//! Functionality for determining who is calling an ABI method
 
-// this can be a generic option when options land
-enum Caller {
-    Some: b256,
-    None: (),
+use ::result::Result;
+use ::address::Address;
+use ::contract_id::ContractId;
+
+// Ref: https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/opcodes.md#gm-get-metadata
+const GM_IS_CALLER_EXTERNAL = 1; // Get if caller is external.
+const GM_GET_CALLER = 2; // Get caller's contract ID.
```

Comment:
Nice. Can you turn this into an enum rather than loose constants?

Reply:
hmm. I can create an enum like this:
```
pub enum GmOptions {
    IsCallerExternal: u8, // Get if caller is external.
    GetCaller: u8, // Get caller's contract ID.
}
```
but I'm bot sure if I can assign actual values, unless I do this as well:
```
let GmOne = GmOptions::IsCallerExternal(1);
let GmTwo = GmOptions::GetCaller(2);
```
Am I missing something here? Using the  `const` approach almost seems cleaner here...

Reply:
This approach in Rust could work once we get enum matching:
```
enum MyEnum {
    A,
    B,
}

impl MyEnum {
    fn value(&self) -> i32 {
        match *self {
            MyEnum::A => 123,
            MyEnum::B => 456,
        }
    }
}
// call like some_myenum_value.value()

```

Reply:
Yeah you don't have to do a `MEQ`, you should be able to compare which variant it is (since that's itself an implicit field).

Reply:
cc @sezna for how we'd do enums with numerical representations.

Reply:
@adlerjohn 
We can't use `match` with enums yet, so if I understand your request here correctly, this will have to wait.
Or, we move forward using  `const`s but add a todo for this.
As @otrho has merged changes to the `B512` type we should probably tag a new release for `lib-std`, and maybe we can get this in there too?

Reply:
Sounds good. TODO it is.